pydm/widgets/timeplot.py

```diff
@@ -231,24 +364,68 @@ def setCurves(self, new_list):
 
     curves = Property("QStringList", getCurves, setCurves)
 
+    def findCurve(self, pv_name):
+        for curve in self._curves:
+            if curve.address == pv_name:
+                return curve
+
+    def refreshCurve(self, curve):
+        """
+        Remove a curve currently being plotted on the timeplot, then redraw that curve, which could have been updated
+        with a new symbol, line style, line width, etc.
+        :param curve:
+        :return:
+        """
+        curve = self.findCurve(curve.channel)
+        if curve:
+            self.removeYChannel(curve)
+            self.addYChannel(y_channel=curve.address, color=curve.color, name=curve.address,
+                             lineStyle=curve.lineStyle, lineWidth=curve.lineWidth, symbol=curve.symbol,
+                             symbolSize=curve.symbolSize)
+
+    def annotateCurve(self, curve, annotation):
+        curve_point = CurvePoint(curve)
+        self.plotItem.addItem(curve_point)
+        annotation.setParentItem(curve_point)
+        curve_point.setPos(1)
+
+    def addLegendItem(self, item, pv_name, force_show_legend=False):
+        self._legend.addItem(item, pv_name)
+        self.setShowLegend(force_show_legend)
+
+    def removeLegendItem(self, pv_name):
+        self._legend.removeItem(pv_name)
+        if len(self._legend.items) == 0:
+            self.setShowLegend(False)
+
     def getBufferSize(self):
         return int(self._bufferSize)
 
     def setBufferSize(self, value):
         if self._bufferSize != int(value):
-            self._bufferSize = max(int(value), 1)
+            # Originally, the bufferSize is the max between the user's input and 1, and 1 doesn't make sense.
+            # So, I'm comparing the user's input with the minimum buffer size, and pick the max between the two
+            self._bufferSize = max(int(value), MINIMUM_BUFFER_SIZE)
             for curve in self._curves:
                 curve.setBufferSize(value)
 
     def resetBufferSize(self):
-        if self._bufferSize != 1200:
-            self._bufferSize = 1200
+        if self._bufferSize != MINIMUM_BUFFER_SIZE:
+            self._bufferSize = MINIMUM_BUFFER_SIZE
             for curve in self._curves:
                 curve.resetBufferSize()
 
     bufferSize = Property("int", getBufferSize,
                           setBufferSize, resetBufferSize)
 
+    def pausePlotting(self):
+        remaining_time = self.redraw_timer.remainingTime()
+        if remaining_time > 0:
```

Comment:
You can query the `QTimer` for the `active` property instead. It will give you a better result.
http://doc.qt.io/qt-5/qtimer.html#active-prop

Reply:
Also, this Pause Plotting logic would be better at the `BasePlot`, right? As it can be beneficial to more plot types.

Reply:
If it is not solved yet through a commit please don't mark as `resolved`.

Reply:
Moved plotting pause to BasePlot.